### PR TITLE
T393

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

File: Sources/Search/OASearchResult.mm [hunk 1]
```diff
@@ -45,9 +45,49 @@ - (double) unknownPhraseMatchWeight
...
+    
+    NSString *fw = [_requiredSearchPhrase getFirstUnknownSearchWord];
+    NSMutableArray<NSString *> *ow = [_requiredSearchPhrase getUnknownSearchWords];
+    
+    if(fw != nil){
+        [searchPhraseNames addObject:fw];
+    }
+    if(ow != nil){
+        [searchPhraseNames addObjectsFromArray:ow];
+    }
```
Comment (nnngrach) [inline Sources/Search/OASearchResult.mm:61]: if(ow)
	[searchPhraseNames addObjectsFromArray:ow];

File: Sources/Search/OASearchResult.mm [hunk 2]
```diff
@@ -45,9 +45,49 @@ - (double) unknownPhraseMatchWeight
...
+                idxMatchedWord = i;
+                break;
+            }
+        }
+        if (!wordMatched) {
+            allWordsMatched = false;
+            break;
+        }
+    }
+    if (_objectType == POI_TYPE) {
```
Comment (nnngrach) [inline Sources/Search/OASearchResult.mm:81]: if ()
{

File: Sources/Search/OASearchResult.mm [hunk 3]
```diff
@@ -45,9 +45,49 @@ - (double) unknownPhraseMatchWeight
...
+        allWordsMatched = false;
+    }
+    double res = allWordsMatched ? [OAObjectType getTypeWeight:_objectType]*10 : [OAObjectType getTypeWeight: UNDEFINED];
+    
+    if ([_requiredSearchPhrase getUnselectedPoiType] != nil) {
```
Comment (nnngrach) [inline Sources/Search/OASearchResult.mm:86]: if ([_requiredSearchPhrase getUnselectedPoiType]) 
{

File: Sources/Search/OASearchPhrase.h [hunk 4]
```diff
@@ -106,5 +106,11 @@ typedef NS_ENUM(NSInteger, EOASearchPhraseDataType)
 
 + (NSComparisonResult) icompare:(int)x y:(int)y;
 
++ (OAPOIBaseType*) unselectedPoiType;
+- (void) splitWords:(NSString *)w ws:(NSMutableArray<NSString *> *)ws;
+
+- (OAPOIBaseType *) getUnselectedPoiType;
+- (void) setUnselectedPoiType:(OAPOIBaseType *) unselectedPoiType;
```
Comment (PaulStets) [inline Sources/Search/OASearchPhrase.h:113]: Extra space before parameter name

File: Sources/Search/OASearchPhrase.mm [hunk 5]
```diff
@@ -280,6 +282,19 @@ - (int) countWords:(NSString *)word
     return cnt;
 }
 
+- (void) splitWords:(NSString *)w
+                 ws:(NSMutableArray<NSString *> *)ws
+{
+    if (w && w.length > 0)
+    {
+        NSArray<NSString *> *wrs = [w componentsSeparatedByRegex:ALLDELIMITERS];
+        for (int i = 0; i < [wrs count]; i++){
```
Comment (PaulStets) [inline Sources/Search/OASearchPhrase.mm:291]: Formatting

File: Sources/Search/OASearchPhrase.mm [hunk 6]
```diff
@@ -405,6 +420,16 @@ - (NSString *) getFirstUnknownSearchWord
     return _firstUnknownSearchWord;
 }
 
+- (OAPOIBaseType *) getUnselectedPoiType
```
Comment (PaulStets) [inline Sources/Search/OASearchPhrase.mm:423]: In obj c, we use ```@property``` for getters and setters.

File: Sources/Search/OASearchResult.mm [hunk 7]
```diff
@@ -45,18 +46,66 @@ - (double) unknownPhraseMatchWeight
...
+    NSString *name = _alternateName ? _alternateName : _localeName;
+    NSMutableArray<NSString *> *localResultNames = [NSMutableArray array];
+    NSMutableArray<NSString *> *searchPhraseNames = [NSMutableArray array];
+    [_requiredSearchPhrase splitWords:name ws:localResultNames];
+    
+    NSString *fw = [_requiredSearchPhrase getFirstUnknownSearchWord];
+    NSMutableArray<NSString *> *ow = [_requiredSearchPhrase getUnknownSearchWords];
+    
+    if (fw)
+    {
```
Comment (PaulStets) [inline Sources/Search/OASearchResult.mm:58]: No need for braces in one-line if conditions

File: Sources/Search/OASearchResult.mm [hunk 8]
```diff
@@ -45,18 +46,66 @@ - (double) unknownPhraseMatchWeight
...
+        {
+            allWordsMatched = NO;
+            break;
+        }
+    }
+    if (_objectType == POI_TYPE)
+    {
+        allWordsMatched = NO;
+    }
+    double res = allWordsMatched ? [OAObjectType getTypeWeight:_objectType]*10 : [OAObjectType getTypeWeight: UNDEFINED];
```
Comment (PaulStets) [inline Sources/Search/OASearchResult.mm:92]: formatting

File: Sources/Search/OASearchResult.mm [hunk 9]
```diff
@@ -45,18 +46,66 @@ - (double) unknownPhraseMatchWeight
...
+    
+    if (_parentSearchResult)
         res = res + [_parentSearchResult getSumPhraseMatchWeight] / MAX_TYPE_WEIGHT;
     
     return res;
 }
 
 - (int) getDepth
 {
-    if (_parentSearchResult != nil)
```
Comment (PaulStets) [inline Sources/Search/OASearchResult.mm:59]: unnecessary change
